UKHomeOffice/nmsw-ui
# NMSW-1329 Crew holding page

# Ticket



----

## AC

After upload 

Given I have uploaded a FAL5 for an arrival journey
When we are still waiting for AMG responses
Then I am shown a holding page
And this page has a refresh button

When I click the refresh button
This page reloads

When the AMG responses are all received
And I reload the page
Then a confirmation message is shown
And there is a link to the task details page

From task list link

Given I have previously uploaded a FAL5
And I am on the task list page
When I click the crew link
And we are still waiting for AMG responses
Then I am shown the holding page

When I click the crew link
And all the AMG responses are received
Then I am shown the FAL5 upload page

----

## To test

You should be able to trigger the holding screen by uploading a crew sheet with thousands of people on it

ARRIVAL

- upload a FAL5
- click save and continue
- check the network tab for `/v1/declaration/<declarationId>?status_count=True`
- if `"Check-in Requested"` is greater than 0
- > you should see a holding screen with a refresh button
- click refresh
- page should reload
- if `crewCount === DC Pending`
- > you should see the task list page

 

FROM TASK LIST PAGE

- click the link for crew
- if `"Check-in Requested"` is greater than 0
- > you should see the holding screen
- if `crewCount === DC Pending`
- > you should see the FAL 5 upload page

----

## Notes
we have updated the content on the passenger holding page to bring it in line with the crew holding page
I've added placeholders for when we have the error responses for the holding page and commented it all out for now
